crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -155,10 +170,62 @@ pub fn extract_meshes(
     commands.insert_or_spawn_batch(not_caster_values);
 }
 
+#[derive(Component)]
+pub struct SkinnedMeshJoints {
+    pub buffer: BufferVec<Mat4>,
+}
+
+impl SkinnedMeshJoints {
+    #[inline]
+    pub fn build(
+        skin: &SkinnedMesh,
+        inverse_bindposes: &Assets<SkinnedMeshInverseBindposes>,
+        joints: &Query<&GlobalTransform>,
+        max_joint_count: usize,
+    ) -> Option<Self> {
+        let inverse_bindposes = inverse_bindposes.get(&skin.inverse_bindposes)?;
+        let bindposes = inverse_bindposes.iter();
+        let skin_joints = skin.joints.iter();
+        let mut buffer = BufferVec::new(BufferUsages::UNIFORM);
+        for (inverse_bindpose, joint) in bindposes.zip(skin_joints).take(max_joint_count) {
+            let joint_matrix = joints.get(*joint).ok()?.compute_matrix();
+            buffer.push(joint_matrix * *inverse_bindpose);
+        }
+        Some(Self { buffer })
+    }
+}
+
+pub fn extract_skinned_meshes(
+    mut commands: Commands,
+    mut previous_len: Local<usize>,
+    query: Query<(Entity, &ComputedVisibility, &SkinnedMesh)>,
+    inverse_bindposes: Res<Assets<SkinnedMeshInverseBindposes>>,
+    joint_query: Query<&GlobalTransform>,
+    wgpu_settings: Res<WgpuSettings>,
+) {
+    let mut values = Vec::with_capacity(*previous_len);
+    let max_joint_count =
+        (wgpu_settings.limits.max_uniform_buffer_binding_size as usize) / JOINT_SIZE;
```

Comment:
We shouldn't be reading `max_uniform_buffer_binding_size` from WgpuSettings. Those are the "requested limits". `RenderDevice` is our source of truth for "actual runtime limits".